contentcuration/contentcuration/viewsets/file.py

```diff
@@ -162,26 +197,18 @@ def delete_from_changes(self, changes):
 
     @action(detail=False, methods=["post"])
     def upload_url(self, request):
-        try:
-            size = request.data["size"]
-            checksum = request.data["checksum"]
-            filename = request.data["name"]
-            file_format = request.data["file_format"]
-            preset = request.data["preset"]
-        except KeyError:
-            return HttpResponseBadRequest(
-                reason="Must specify: size, checksum, name, file_format, and preset"
-            )
-
-        duration = request.data.get("duration")
-        if duration is not None:
-            if not isinstance(duration, (int, float)):
-                return HttpResponseBadRequest(reason="File duration must be a number")
-            duration = math.floor(duration)
-            if duration <= 0:
-                return HttpResponseBadRequest(
-                    reason="File duration is equal to or less than 0"
-                )
+        # Validate input using the new serializer
+        serializer = FileUploadURLSerializer(data=request.data)
+        if not serializer.is_valid():
```

Comment:
Can also just call:
```
serializer.is_valid(raise_exception=True)
```

https://www.django-rest-framework.org/api-guide/serializers/#validation

This has the advantage of formatting the errors into the consistent response object format.